sdk/core/Azure.Core/src/Pipeline/HttpPipeline.cs

```diff
@@ -55,6 +69,22 @@ public HttpMessage CreateMessage()
             return new HttpMessage(CreateRequest(), ResponseClassifier);
         }
 
+        /// <summary>
+        /// Creates a new <see cref="HttpMessage"/> instance.
+        /// </summary>
+        /// <param name="context">Context specifying the message options.</param>
+        /// <returns>The message.</returns>
+        public HttpMessage CreateMessage(RequestContext context)
+        {
+            var message = CreateMessage();
+            if (context != null)
+            {
+                message.CancellationToken = context.CancellationToken;
```

Comment:
What if the message already has a token? Should we combine?

Reply:
We know in this method that it doesn't because it was just created.  But I'll take a closer look at the CancellationToken flow here to make sure it makes sense.

Reply:
I believe CreateMessage() creates a new message without a CancellationToken, but I'll go back and doublecheck this.

Reply:
If `HttpMessage` instance is created directly in this method, you can set `cancellationToken` and `policies` in constructor.

Reply:
CreateMessage is virtual. Custom transports can create one with a CT.

Reply:
> CreateMessage is virtual. Custom transports can create one with a CT.

Ah, thank you!

Reply:
Actually, @KrzysztofCwalina, CreateMessage() is not virtual: https://github.com/Azure/azure-sdk-for-net/pull/25214/files#diff-d8fef9e5888142bd6d9a5398c2d61a54cde4d38e79fdd50a204650a1cf44abaaR767

Reply:
Ah, you are right. I was thinking about the virtual method on transport, but it only creates the request, not the whole message.

Reply:
I removed the update to the CancellationToken anyway - no need to add another place it's set on the message to keep track of.